common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -923,4 +1076,12 @@ public static String getClosestSuggestionsOnInvalidName(
 
     return String.join(", ", suggestions);
   }
+
+  public static List<CollationIdentifier> listCollations(String catalog, String schema) {
+    return Collation.CollationSpec.listCollations(catalog, schema);
+  }
+
+  public static Collation loadCollation(CollationIdentifier collationIdentifier) {
```

Comment:
I would say we need to avoid passing Collation object anywhere for default collations. The problem is that we are storing these in the hashmap when loadCollation is called and we would just overload the map. Could we maybe create a record like you did and pass that as a result. Also I do not think we need to add schema and catalog to Collation in the current implementation, as show collations has that information from before.

Reply:
Do you mean: we define as follows ?
```java
public record CollationMeta(...)
 CollationMeta loadCollation(CollationIdentifier collationIdentifier)
```

Reply:
Done.